src/master/sdks-api.md

```diff
@@ -1845,6 +1847,182 @@ If you're using another development language, please read up on our [immugw](htt
 
 <br/>
 
+## Streams
+Immudb provides stream capabilities.
+Internally it uses “delimited” messages technique, every chunk has a trailer that describe the length of the message. In this way the receiver can recompose chunk by chunk the original payload.
+Stream methods accepts a `readers` as a part of input and output arguments. In this way the large value is decomposed in small chunks that are streamed over the wire.
+Client don't need to allocate the entire value when sending and can read the received one progressively.
+For example a client could send a large file much greater than available ram memory.
+> At the moment `immudb` is not yet able to write the data without allocating the entire received object, but in the next release it will be possible a complete communication without allocations.
+  The maximum size of a transaction sent with streams is temporarily limited to a payload of 32M.
+
+Supported stream method now available in the SDK are:
+
+	StreamSet
+	StreamGet
+	StreamVerifiedSet
+	StreamVerifiedGet
+	StreamScan
+	StreamZScan
+	StreamHistory
+	StreamExecAll
+
+:::: tabs
+
+::: tab Go
+Here an example on how to send a large file and a regular key value to immudb.
+
+It's possible to specify the chunk size of the stream with `WithStreamChunkSize()` method.
+
+```go
+    client, err := immuclient.NewImmuClient(immuclient.DefaultOptions().WithStreamChunkSize(4096))
+	if err != nil {
+		log.Fatal(err)
+	}
+    ctx := context.Background()
+    lr, err := client.Login(ctx, []byte(`immudb`), []byte(`immudb`))
+    if err != nil {
+        log.Fatal(err)
+    }
+    md := metadata.Pairs("authorization", lr.Token)
+    ctx = metadata.NewOutgoingContext(context.Background(), md)
+    myFileName := "streams.go"
+    key1 := []byte("key1")
+    val1 := []byte("val1")
+    f, err := os.Open(myFileName)
+    if err != nil {
+        log.Fatal(err)
+    }
+    stats, err := os.Stat(myFileName)
+    if err != nil {
+        log.Fatal(err)
+    }
+
+    kv1 := &stream.KeyValue{
+        Key: &stream.ValueSize{
+            Content: bytes.NewBuffer(key1),
+            Size:    len(key1),
+        },
+        Value: &stream.ValueSize{
+            Content: bytes.NewBuffer(val1),
+            Size:    len(val1),
+        },
+    }
+    kv2 := &stream.KeyValue{
+        Key: &stream.ValueSize{
+            Content: bytes.NewBuffer([]byte(myFileName)),
+            Size:    len(myFileName),
+        },
+        Value: &stream.ValueSize{
+            Content: f,
+            Size:    int(stats.Size()),
+        },
+    }
+
+    kvs := []*stream.KeyValue{kv1, kv2}
+    _, err = client.StreamSet(ctx, kvs)
+    if err != nil {
+        log.Fatal(err)
+    }
+    entry, err := client.StreamGet(ctx, &schema.KeyRequest{ Key: []byte(myFileName)})
+    if err != nil {
+        log.Fatal(err)
+    }
+    fmt.Printf("returned key %s", entry.Key)
+```
+:::
+
+::: tab Java
+This feature is not yet supported or not documented.
```

Comment:
I think we can say something very explicit here: "Streams is not supported yet in this language SDK".

Reply:
We use that placeholder in every place of the doc. To avoid to add a custom message isn't better to keep as it is or you suggest to change it in every place?